Implement transform for full single-qubit gate fusion (#1458)
PennyLaneAI/pennylane: .github/CHANGELOG.md

<h3>New features since last release</h3>

> * A new quantum function has been added to perform full fusion of

Author: ```suggestion
* A new quantum function transform has been added to perform full fusion of
```

  arbitrary-length sequences of single-qubit gates.

>   [(#1458)](https://github.com/PennyLaneAI/pennylane/pull/1455)

Author: ```suggestion
  [(#1458)](https://github.com/PennyLaneAI/pennylane/pull/1458)
```

  The `single_qubit_fusion` transform acts on all sequences of

>   single-qubit operations in a quantum function, and converts the
>   sequences to a single `Rot` gate.. For example given the circuit:

Author: ```suggestion
  single-qubit operations in a quantum function, and converts each
  sequence to a single `Rot` gate. For example given the circuit:
```
